### PR TITLE
feat(channel): add default forum layout

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

#936

## Checklist

- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/types/channel.py [hunk 1]
```diff
@@ -63,6 +64,7 @@ class ForumChannel(_ThreadedBaseChannel):
     default_reaction_emoji: NotRequired[Optional[DefaultReaction]]
     default_thread_rate_limit_per_user: NotRequired[int]
     available_tags: NotRequired[List[ForumTag]]
+    default_forum_layout: ForumLayoutType
```
Comment (EmreTech) [inline nextcord/types/channel.py:67]: According to the relevant [Discord API Docs PR](https://github.com/discord/discord-api-docs/pull/5693), this attribute is optional.

File: nextcord/channel.py [hunk 2]
```diff
@@ -1058,6 +1068,7 @@ async def edit(
         flags: ChannelFlags = ...,
         reason: Optional[str] = ...,
         default_sort_order: Optional[SortOrderType] = ...,
+        default_forum_layout: Optional[ForumLayoutType] = ...,
```
Comment (EmreTech) [inline nextcord/channel.py:1071]: Unlike `default_sort_order`, this is not nullable. It is strictly an integer (in this case, a forum layout type).

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: docs/api.rst [hunk 3]
```diff
@@ -3037,6 +3037,24 @@ of :class:`enum.Enum`.
 
         Sort forum posts by their creation date.
 
+.. class:: ForumLayoutType
+
+    .. versionadded:: 2.4
+
+    The default layout type used to display posts in a :class:`ForumChannel`.
```
Comment (ooliver1) [inline docs/api.rst:3044]: Usually the first line of docs is the short summary.
```suggestion
    The default layout type used to display posts in a :class:`ForumChannel`.

    .. versionadded:: 2.4
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by ooliver1** [comment]
Could this be updated to the naming and nullability shown in discord/discord-api-docs#5745 - no idea why they just superseeded it.

File: docs/api.rst (hunk 3 above)
Comment (Soheab) [reply docs/api.rst:3044]: I could sneak the same thing in for `SortOrderType` if you want.

**Review by Skelmis — APPROVED** [review]

File: nextcord/channel.py [hunk 4]
```diff
@@ -941,6 +947,10 @@ def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
             self.default_sort_order: Optional[SortOrderType] = try_enum(SortOrderType, sort_order)
         else:
             self.default_sort_order: Optional[SortOrderType] = None
+        if layout := data.get("default_forum_layout"):
```
Comment (ooliver1) [inline nextcord/channel.py:950]: The docs only mention it as ommitable because it doesn't exist on other channel types. Could this default to `not_set`?

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/channel.py [hunk 5]
```diff
@@ -883,6 +884,10 @@ class ForumChannel(abc.GuildChannel, Hashable):
         The default sort order type used to sort posts in forum channels.
 
         .. versionadded:: 2.3
+    default_forum_layout: Optional[:class:`ForumLayoutType`]
```
Comment (ooliver1) [inline nextcord/channel.py:887]: Docs need updating.
```suggestion
    default_forum_layout: :class:`ForumLayoutType`
```

File: nextcord/enums.py [hunk 6]
```diff
@@ -711,6 +711,12 @@ class SortOrderType(IntEnum):
     creation_date = 1
 
 
+class ForumLayoutType(IntEnum):
```
Comment (ooliver1) [inline nextcord/enums.py:714]: Needs to be added to `__all__`

File: nextcord/channel.py [hunk 7]
```diff
@@ -1058,6 +1068,7 @@ async def edit(
         flags: ChannelFlags = ...,
         reason: Optional[str] = ...,
         default_sort_order: Optional[SortOrderType] = ...,
+        default_forum_layout: ForumLayoutType = ...,
```
Comment (ooliver1) [inline nextcord/channel.py:1071]: Missing from http.py `HTTPClient.edit_channel`->`valid_keys`, so cannot be set.

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Review by ooliver1 — APPROVED** [review]
Tested
```py
@bot.slash_command()
async def test(inter: Interaction):
    # Why would Discord not allow you to set on creation?
    assert inter.guild is not None
    forum_channel = await inter.guild.create_forum_channel("test", topic="test2")

    assert forum_channel.default_forum_layout is ForumLayoutType.not_set
    forum_channel = await forum_channel.edit(default_forum_layout=ForumLayoutType.gallery)
    assert (
        forum_channel.default_forum_layout is ForumLayoutType.gallery
    ), forum_channel.default_forum_layout

    await forum_channel.delete()

    await inter.response.send_message("Done")
```

**Review by EmreTech — APPROVED** [review]

**Pull request merged**